readthedocs/projects/models.py

```diff
@@ -772,22 +785,38 @@ def has_media(self, type_, version_slug=LATEST):
         if settings.RTD_BUILD_MEDIA_STORAGE:
             storage = get_storage_class(settings.RTD_BUILD_MEDIA_STORAGE)()
             storage_path = self.get_storage_path(
-                type_=type_, version_slug=version_slug
+                type_=type_, version_slug=version_slug,
+                version_type=version_type
             )
             return storage.exists(storage_path)
 
         return False
 
-    def has_pdf(self, version_slug=LATEST):
-        return self.has_media(MEDIA_TYPE_PDF, version_slug=version_slug)
+    def has_pdf(self, version_slug=LATEST, version_type=None):
```

Comment:
For another refactor, in the future: it seems that these methods are more tied to the `Version` object instead of the `Project`. Actually, the method from the Project look that are tied to the `LATEST` version, which makes sense to me.